src/compiler/builder.ts

```diff
@@ -451,6 +451,11 @@ function createBuilderProgramState(newProgram: Program, oldState: Readonly<Reusa
                     pendingEmitKind;
             }
         }
+        else if (!canCopySemanticDiagnostics) {
+            // No emit being done, but semantic diagnostics must be recalculated - since these (and the options that change these)
+            // are in the build info file, we need to emit it to keep the info on-disk up-to-date
+            state.buildInfoEmitPending = true;

```

Comment:
This should be set only if (!outputFile)

Reply:
As mentioned somewhere below, you want this condition to be `useOldState && !!oldState!.compilerOptions.noCheck !== !!compilerOptions.noCheck`

Reply:
I have no clue why this would be `noCheck`-specific, though? `noCheck` seems to me to be no different from any other compiler option that varies the errors we produce.